roles/azure_virtual_machine_with_public_ip/tasks/create.yml

```diff
@@ -115,7 +131,7 @@
     vm_size: "{{ azure_virtual_machine_with_public_ip_vm.size | default(omit) }}"
     network_interfaces: "{{ azure_virtual_machine_with_public_ip_vm.network_interfaces | default(vm_name) }}"
     os_type: "{{ azure_virtual_machine_with_public_ip_vm.os | default(omit) }}"
-    availability_set: "{{ azure_virtual_machine_with_public_ip_vm.availability_set | default(omit) }}"
+    availability_set: "{{ azure_virtual_machine_with_public_ip_availability_set.name | default(omit) }}"
```

Comment:
If the object is omitted can we still reference a child (.name)?  I know you have the default(omit) but I forgot how ansible handles this.  Did you test this with the object omitted?

Reply:
I ran a test without azure_virtual_machine_with_public_ip_availability_set and it passed, so yes, this construction can handle non-existent parent :)

Reply:
I'm having second thought about it, even if test passed. It is cost nothing to use 'azure_virtual_machine_with_public_ip_availability_set' as   condition and it won't cause any questions. 
I'll replace 'azure_virtual_machine_with_public_ip_availability_set.name' with 'azure_virtual_machine_with_public_ip_availability_set '

Reply:
fixed